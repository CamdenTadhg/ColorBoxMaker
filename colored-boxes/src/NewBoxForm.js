import react, {useState} from 'react';
import './NewBoxForm.css'

const NewBoxForm = ({addBox}) => {
    const initialState = {
        color: "",
        width: 0,
        height: 0
    };
    const [formData, setFormData] = useState(initialState);
    const handleChange = (event) => {
        const {name, value} = event.target;
        setFormData(fData => ({
            ...fData,
            [name]: value
        }));
    };

    const handleSubmit = (event) => {
        event.preventDefault();
        const {color, width, height} = formData;
        addBox(color, width, height);
        setFormData(initialState);
    };

    return (
        <form className="NewBoxForm" onSubmit={handleSubmit} data-testid='newBoxForm'>
            <label className="NewBoxForm-label" htmlFor="color">Color:
                <input type="text" id="color" name="color" placeholder="color" value={formData.color} onChange={handleChange}/>
            </label>
            <label className="NewBoxForm-label" htmlFor="width">Width: 
                <input type="number" id="width" name="width" placeholder="width" value={formData.width} onChange={handleChange}/>
            </label>
            <label className="NewBoxForm-label" htmlFor="height">Height:
                <input type="number" id="height" name="height" placeholder="height" value={formData.height} onChange={handleChange}/>
            </label>
            <button>Add Box</button>
        </form>
    )
};

export default NewBoxForm;
