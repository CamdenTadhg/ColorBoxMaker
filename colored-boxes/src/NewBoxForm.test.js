import react from 'react';
import {render} from '@testing-library/react';
import NewBoxForm from './NewBoxForm.js';

it('should render correctly', function(){
    render(<NewBoxForm/>);
});

it('should match snapshot', function() {
    const {asFragment} = render(<NewBoxForm/>);
    expect(asFragment()).toMatchSnapshot();
});

