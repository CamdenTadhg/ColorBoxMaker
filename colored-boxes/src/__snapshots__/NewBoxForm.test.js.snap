// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewBoxForm"
    data-testid="newBoxForm"
  >
    <label
      class="NewBoxForm-label"
      for="color"
    >
      Color:
      <input
        id="color"
        name="color"
        placeholder="color"
        type="text"
        value=""
      />
    </label>
    <label
      class="NewBoxForm-label"
      for="width"
    >
      Width:
      <input
        id="width"
        name="width"
        placeholder="width"
        type="number"
        value="0"
      />
    </label>
    <label
      class="NewBoxForm-label"
      for="height"
    >
      Height:
      <input
        id="height"
        name="height"
        placeholder="height"
        type="number"
        value="0"
      />
    </label>
    <button>
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
